keras_rcnn/layers/losses.py

```diff
@@ -1,45 +1,38 @@
 import keras.layers
 
+import keras_rcnn.backend
 
-class Classification(keras.layers.Layer):
+
+class ClassificationLoss(keras.layers.Layer):
     def __init__(self, anchors, **kwargs):
         self.anchors = anchors
 
-        self.is_placeholder = True
-
-        super(Classification, self).__init__(**kwargs)
+        super(ClassificationLoss, self).__init__(**kwargs)
 
-    def _loss(self, y_true, y_pred):
-        # Binary classification loss
-        x, y = y_pred[:, :, :, :], y_true[:, :, :, self.anchors:]
+    def _loss(self, rpn_labels, rpn_classification):
+        rpn_classification = keras.backend.reshape(rpn_classification, [-1, 2])
+        rpn_classification = keras.backend.gather(rpn_classification, keras_rcnn.backend.where(keras.backend.not_equal(rpn_labels, -1)))
+        rpn_labels         = keras.backend.gather(rpn_labels, keras_rcnn.backend.where(keras.backend.not_equal(rpn_labels, -1)))
```

Comment:
From @yhenon (on Slack), author of the original loss:

> @hgaiser it looks nice but you should explicitly handle the case where `keras.backend.not_equal(rpn_labels, -1)` returns an empty array (it can happen, especially if rpn_max_proposals is low) - otherwise you will get NaNs from the `mean()` calculation (edited)